signer/xpi/xpi.go

```diff
@@ -556,16 +557,70 @@ func VerifySignedFile(signedFile signer.SignedFile, truststore *x509.CertPool, o
 		return errors.Wrap(err, "xpi: error verifying PKCS7 signature for signed file")
 	}
 
-	if len(opts.COSEAlgorithms) > 0 {
-		err = verifyCOSEManifest(signedFile)
-		if err != nil {
-			return errors.Wrap(err, "xpi: error verifying COSE manifest for signed file")
-		}
-		err = verifyCOSESignatures(signedFile, truststore, opts)
-		if err != nil {
-			return errors.Wrap(err, "xpi: error verifying COSE signatures for signed file")
+	coseSig, err := readFileFromZIP(signedFile, coseSigPath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "xpi: no COSE signature found, skipping")
```

Comment:
nit: worth pulling this through L590 out into a `readOptionsFromCOSESig(signedFile signer.SignedFile) (opts Options, err error)` function to have it be at the same level of abstraction as the other fns called from this function

nit: For calls from autograph-client and unit tests where signing with COSE doesn't write a `cose.sig` this will fail silently, which is undesirable in IMO and should, at a minimum, be documented somewhere.

Reply:
I fixed the first nit by moving it to `xpi.ReadCOSEAlgsFromSig`. Not sure what to do about the second one... I did put a comment in there saying we fail silently for COSE, like Firefox does.